io/directory.go

```diff
@@ -203,11 +229,26 @@ func (d *BasicDirectory) Find(ctx context.Context, name string) (ipld.Node, erro
 
 // RemoveChild implements the `Directory` interface.
 func (d *BasicDirectory) RemoveChild(ctx context.Context, name string) error {
-	err := d.node.RemoveNodeLink(name)
+	// We need to *retrieve* the link before removing it to update the estimated
+	// size.
+	// FIXME: If this is too much of a potential penalty we could leave a fixed
+	//  CID size estimation based on the most common one used (normally SHA-256).
+	//  Alternatively we could add a GetAndRemoveLink method in `merkledag` to
+	//  iterate node links slice only once.
+	link, err := d.node.GetNodeLink(name)
```

Comment:
If traversing this list becomes a problem, a factor of 2 isn't going to make much of a difference. We'd likely need to cache a link resolution map in that case.